script/tool/lib/src/pubspec_check_command.dart

```diff
@@ -0,0 +1,178 @@
+      }
+    }
+
+    if (failingPackages.isNotEmpty) {
+      print('The following packages have pubspec issues:');
+      for (final String package in failingPackages) {
+        print('  $package');
+      }
+      throw ToolExit(1);
+    }
+
+    print('\nNo pubspec issues found!');
+  }
+
+  Future<bool> _checkPubspec(
+    File pubspecFile, {
+    required String packageName,
+  }) async {
+    const String indentation = '  ';
+    final String contents = pubspecFile.readAsStringSync();
+    final Pubspec? pubspec = _tryParsePubspec(contents);
+    if (pubspec == null) {
+      return false;
+    }
+
+    final List<String> pubspecLines = contents.split('\n');
+    final List<String> sectionOrder = pubspecLines.contains('  plugin:')
+        ? _majorPluginSections
+        : _majorPackageSections;
+    bool passing = _checkSectionOrder(pubspecLines, sectionOrder);
+    if (!passing) {
+      print('${indentation}Major sections should follow standard '
+          'repository ordering:');
+      const String listIndentation = '$indentation$indentation';
+      print('$listIndentation${sectionOrder.join('\n$listIndentation')}');
+    }
+
+    if (pubspec.publishTo != 'none') {
+      final List<String> repositoryErrors =
+          _checkForRepositoryLinkErrors(pubspec, packageName: packageName);
+      if (repositoryErrors.isNotEmpty) {
+        for (final String error in repositoryErrors) {
+          print('$indentation$error');
+        }
+        passing = false;
+      }
+
+      if (!_checkIssueLink(pubspec)) {
+        print(
+            '${indentation}A package should have an "issue_tracker" link to a '
+            'search for open flutter/flutter bugs with the relevant label:\n'
+            '$indentation$indentation$_expectedIssueLinkFormat<package label>');
+        passing = false;
+      }
+    }
+
+    return passing;
+  }
+
+  Pubspec? _tryParsePubspec(String pubspecContents) {
+    try {
+      return Pubspec.parse(pubspecContents);
+    } on Exception catch (exception) {
+      print('  Cannot parse pubspec.yaml: $exception');
+    }
+    return null;
+  }
+
+  bool _checkSectionOrder(
+      List<String> pubspecLines, List<String> sectionOrder) {
+    int previousSectionIndex = 0;
+    for (final String line in pubspecLines) {
+      final int index = sectionOrder.indexOf(line);
+      if (index == -1) {
+        continue;
+      }
+      if (index < previousSectionIndex) {
+        return false;
+      }
+      previousSectionIndex = index;
+    }
+    return true;
+  }
+
+  List<String> _checkForRepositoryLinkErrors(
+    Pubspec pubspec, {
+    required String packageName,
+  }) {
+    final List<String> errorMessages = <String>[];
+    if (pubspec.repository == null) {
+      errorMessages.add('Missing "repository"');
+    } else if (!pubspec.repository!.path.endsWith(packageName)) {
+      errorMessages
+          .add('The "repository" link should end with the package name.');
```

Comment:
Sadly I couldn't make it check for exact correctness without making things a lot more complicated, due to the multiple repos this runs in, but this will prevent both just putting the base repo and also copy pasta errors.

Reply:
Nah, this is great!